safe/common/resource_parameter_widget.py

```diff
@@ -21,12 +21,23 @@ def __init__(self, parameter, parent=None):
         .. versionadded:: 2.3
 
         :param parameter: A ResourceParameter object.
-        :type parameter: ResourceParameter
+        :type parameter: ResourceParameter, FloatParameter
 
         """
+        # pylint: disable=E1002
         super(ResourceParameterWidget, self).__init__(parameter, parent)
+        # pylint: enable=E1002
         self.set_unit()
 
+    def get_parameter(self):
+        """Obtain the parameter object from the current widget state.
+
+        :returns: A BooleanParameter from the current state of widget
+
```

Comment:
:rtype: BooleanParameter

If you use rtype, pycharm will add extra syntax checking and syntax awareness to code completion for you.